core/discovery/api/finder.go

```diff
@@ -50,12 +57,12 @@ func (finder *finderAPI) GetProposal(id market.ProposalID) (*market.ServicePropo
 	return &proposals[0], nil
 }
 
-// FindProposals fetches currently active service proposals from discovery
-func (finder *finderAPI) FindProposals(filter market.ProposalFilter) ([]market.ServiceProposal, error) {
-	return finder.mysteriumAPI.QueryProposals(mysterium.ProposalsQuery{
-		NodeKey:            filter.ProviderID,
-		ServiceType:        filter.ServiceType,
-		AccessPolicyID:     filter.AccessPolicy.ID,
-		AccessPolicySource: filter.AccessPolicy.Source,
-	})
+// FindProposals fetches currently active service proposals from discovery by given filter
+func (finder *finderAPI) FindProposals(filter discovery.ProposalFilter) ([]market.ServiceProposal, error) {
```

Comment:
why not take in FilterQueryable interface as a param instead of having to cast it inside?

Reply:
Because `finderAPI` and `finderStorage` implements same `ProposalFinder` interface and  just several filters are serialisable for `finderAPI`.
Anyway it's temporarily and we'll drop `finderAPI` soon

Reply:
Anyways, I improved it now by moving interfaces to caller side